jolt-core/src/zkvm/spartan/shift.rs

```diff
@@ -398,3 +310,455 @@ impl<F: JoltField, T: Transcript> SumcheckInstance<F, T> for ShiftSumcheck<F> {
         flamegraph.visit_root(self);
     }
 }
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Allocative)]
+enum Prover<F: JoltField> {
+    Phase1(Phase1Prover<F>), // 1st half (prefix-suffix sc)
+    Phase2(Phase2Prover<F>), // 2st half (regular sc)
+}
```

Comment:
We don't use enums for 2 phased prover state anywhere in the codebase. For consistency, maybe just make ShiftSumcheckProverState a struct and merge fields of Phase1Prover and Phase2Prover, especially given that Phase1 stores a bunch of fields for Phase2. As well as move the compute_prover_message and bind logic back to SumcheckInstanceTrait implementation.